File: Contests/ACM CodeJam 18/faceTheBook.cpp
#include <iostream>

using namespace std;

int main()
{
	int t;
	cin >> t;

	long long T, N, X, F;
	long long users;
	for (int i = 0; i < t; i++)
	{
		cin >> T >> N >> X >> F;

		users = N;
		for (int j = 0; j < T; j++)
		{
			N = N / 2 * 3 * X;
			users += N;
			X *= F;
		}

		cout << users << "\n";
	}

	return 0;
}

File: Contests/ACM CodeJam 18/faceTheBook.py
import sys

def main():
	t = int(input())

	for i in range(t):
		T, N, X, F = input().split();

		T = int(T)
		N = int(N)
		X = int(X)
		F = float(F)

		users = N

		for j in range(T):
			N = N * 3 * X / 2
			users += N
			X *= F

		print(int(users))

if __name__ == '__main__':
	main()